priam/src/main/java/com/netflix/priam/backup/AbstractBackup.java

```diff
@@ -117,18 +120,23 @@ protected final void initiateBackup(
             String monitoringFolder, BackupRestoreUtil backupRestoreUtil) throws Exception {
 
         File dataDir = new File(config.getDataFileLocation());
-        if (!dataDir.exists()) {
+        if (!dataDir.exists() || !dataDir.isDirectory()) {
             throw new IllegalArgumentException(
-                    "The configured 'data file location' does not exist: "
+                    "The configured 'data file location' does not exist or is not a directory: "
                             + config.getDataFileLocation());
         }
         logger.debug("Scanning for backup in: {}", dataDir.getAbsolutePath());
-        for (File keyspaceDir : dataDir.listFiles()) {
+        File[] keyspaceDirectories = dataDir.listFiles();
```

Comment:
listFiles returns an array. Iterating over an array without copying it, returns NPE. Ideally, we should be using DirectoryStream when we wish to iterate over results. I will open another PR to remove all the calls to listFiles() where we directly iterate.